server/webhook/merge_request.go

```diff
@@ -29,57 +29,140 @@ func (w *webhook) handleDMMergeRequest(event *gitlab.MergeEvent) ([]*HandleWebho
-						assignedInPrevious = true
-						break
-					}
+			// Handle change in assignees
+			if event.Changes.Assignees.Current != nil || event.Changes.Assignees.Previous != nil {
+				isAuthorPresent, updatedPreviousUsers, updatedCurrentUsers := w.handleDMNotificationUsers(event.ObjectAttributes.AuthorID, event.Changes.Assignees.Previous, event.Changes.Assignees.Current)
+
+				// Check if the MR author is present in the assignee changes
+				if !isAuthorPresent {
+					message = fmt.Sprintf("[%s](%s) updated the list of assignees for the merge request [%s!%v](%s)", senderGitlabUsername, w.gitlabRetreiver.GetUserURL(senderGitlabUsername), event.ObjectAttributes.Target.PathWithNamespace, event.ObjectAttributes.IID, event.ObjectAttributes.URL)
+					handlers = append(handlers, &HandleWebhook{
+						Message: message,
+						ToUsers: []string{authorGitlabUsername},
+						From:    senderGitlabUsername,
+					})
 				}
-				if !assignedInPrevious {
-					toUsers = append(toUsers, w.gitlabRetreiver.GetUsernameByID(currentAssigneeID))
+
+				if len(updatedPreviousUsers) != 0 {
+					message = fmt.Sprintf("[%s](%s) removed you as an assignee from the merge request [%s!%v](%s)", senderGitlabUsername, w.gitlabRetreiver.GetUserURL(senderGitlabUsername), event.ObjectAttributes.Target.PathWithNamespace, event.ObjectAttributes.IID, event.ObjectAttributes.URL)
+					handlers = append(handlers, &HandleWebhook{
+						Message: message,
+						ToUsers: updatedPreviousUsers,
+						From:    senderGitlabUsername,
+					})
+				}
+
+				if len(updatedCurrentUsers) != 0 {
+					message = fmt.Sprintf("[%s](%s) assigned you to merge request [%s!%v](%s)", senderGitlabUsername, w.gitlabRetreiver.GetUserURL(senderGitlabUsername), event.ObjectAttributes.Target.PathWithNamespace, event.ObjectAttributes.IID, event.ObjectAttributes.URL)
+					handlers = append(handlers, &HandleWebhook{
+						Message: message,
+						ToUsers: updatedCurrentUsers,
+						From:    senderGitlabUsername,
+					})
 				}
 			}
 
-			message = fmt.Sprintf("[%s](%s) assigned you to merge request [%s!%v](%s)", senderGitlabUsername, w.gitlabRetreiver.GetUserURL(senderGitlabUsername), event.ObjectAttributes.Target.PathWithNamespace, event.ObjectAttributes.IID, event.ObjectAttributes.URL)
+			// Handle change in reviewers
+			if event.Changes.Reviewers.Current != nil || event.Changes.Reviewers.Previous != nil {
+				isAuthorPresent, updatedPreviousUsers, updatedCurrentUsers := w.handleDMNotificationUsers(event.ObjectAttributes.AuthorID, event.Changes.Reviewers.Previous, event.Changes.Reviewers.Current)
+
+				// Check if the MR author is present in the assignee changes
+				if !isAuthorPresent {
+					message = fmt.Sprintf("[%s](%s) updated the list of reviewers for the merge request [%s!%v](%s)", senderGitlabUsername, w.gitlabRetreiver.GetUserURL(senderGitlabUsername), event.ObjectAttributes.Target.PathWithNamespace, event.ObjectAttributes.IID, event.ObjectAttributes.URL)
```

Comment:
I'm not sure if this is something I would want to get a notification for. Mainly just "assigned you" or "requested your review" like the GitHub plugin works @hanzei What do you think?

Reply:
Sounds good :+1: